lib/action_controller/serialization.rb

```diff
@@ -56,10 +56,15 @@ def use_adapter?
 
     [:_render_option_json, :_render_with_renderer_json].each do |renderer_method|
       define_method renderer_method do |resource, options|
-        options.fetch(:serialization_context) do
-          options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, options)
+        options.freeze
+        serialization_options = options.deep_dup
+        if options.key?(:serializer)
+          serialization_options[:serializer] = options[:serializer]
         end
-        serializable_resource = get_serializer(resource, options)
+        unless serialization_options.key?(:serialization_context)
+          serialization_options[:serialization_context] = ActiveModelSerializers::SerializationContext.new(request, serialization_options)
+        end
```

Comment:
Isn't fetch main use case supposed to be "retrieving an object"? I changed it because for me using fetch when we don't use the return value feels weird. But I can revert this change if needed.